var/spack/repos/builtin/packages/spfft/package.py

```diff
@@ -26,9 +28,10 @@ class Spfft(CMakePackage):
     variant('fortran', default=False, description="enable fortran")
     variant('build_type', default='Release', description='CMake build type',
             values=('Debug', 'Release', 'RelWithDebInfo'))
-    depends_on('fftw')
+    depends_on('fftw-api@3')
     depends_on('mpi', when='+mpi')
-    depends_on('cuda', when='+cuda')
+
+    conflicts('+cuda ^cuda@11:', when='@:0.9.11', msg="CUDA 11 support requires SpFFT 0.9.12")
```

Comment:
I would change this to:
```python
depends_on('cuda@:10', when='@:0.9.11 +cuda')
```
`conflicts` only causes the concretizer to crash, `depends_on` will cause the concretizer to make the right choice (most of the time).

Reply:
Nice, that's a better way to express this indeed